control/controlclient/controlclient_test.go

```diff
@@ -147,3 +151,42 @@ func TestCanSkipStatus(t *testing.T) {
 		t.Errorf("Status fields = %q; this code was only written to handle fields %q", f, want)
 	}
 }
+
+func TestRetryableErrors(t *testing.T) {
+	errors := []struct {
+		err  error
+		want bool
+	}{
+		{errNoNoiseClient, true},
+		{errNoNodeKey, true},
+		{fmt.Errorf("%w: %w", errNoNoiseClient, errors.New("no noise")), true},
+		{fmt.Errorf("%w: %w", errHTTPPostFailure, errors.New("bad post")), true},
+		{fmt.Errorf("%w: %w", errNoNodeKey, errors.New("not node key")), true},
+		{context.Canceled, false},
+		{fmt.Errorf("%w: %w", context.Canceled, errors.New("ctx cancelled")), false},
+		{errBadHTTPResponse(429, "too may requests"), true},
+		{errBadHTTPResponse(500, "internal server eror"), true},
+		{errBadHTTPResponse(502, "bad gateway"), true},
+		{errBadHTTPResponse(503, "service unavailable"), true},
+		{errBadHTTPResponse(504, "gateway timeout"), true},
+		{errBadHTTPResponse(1234, "random error"), false},
+	}
+
+	for _, e := range errors {
+		if !cmp.Equal(isRetryableErrorForTest(e.err), e.want) {
+			t.Fatalf("error evaluator failed for %v", e.err)
+		}
```

Comment:
```suggestion
		if got := isRetryableErrorForTest(e.err); got != e.want {
		    t.Fatalf("retryable(%v): got %v; want %v", e.err, got, want)
		}
```

Also: consider writing test-driven tests: https://go.dev/wiki/TableDrivenTests

Reply:
This appears to be unresolved?